lumigator/python/mzai/backend/backend/tests/conftest.py

```diff
@@ -107,40 +111,79 @@ def db_session(db_engine: Engine):
             session.rollback()
 
 
-@pytest.fixture(scope="session")
-def localstack_container():
-    """Initialize a LocalStack test container."""
-    with LocalStackContainer("localstack/localstack:3.4.0") as localstack:
-        yield localstack
+# @pytest.fixture(scope="session")
+# def localstack_container():
+#     """Initialize a LocalStack test container."""
+#     with LocalStackContainer("localstack/localstack:3.4.0") as localstack:
+#         yield localstack
 
 
-@pytest.fixture(scope="session", autouse=True)
-def setup_aws(localstack_container: LocalStackContainer):
-    """Setup env vars/AWS resources for use with the LocalStack container."""
+@pytest.fixture(scope="function")
+def fake_s3_storage() -> dict:
+    return dict()
+
+@pytest.fixture(scope="function")
+def fake_s3_client(fake_s3_storage) -> S3Client:
+    os.environ["AWS_ACCESS_KEY_ID"] = "test"
+    # Please check https://github.com/localstack/localstack/issues/5894
+    # for info about the test region used
+    os.environ["AWS_DEFAULT_REGION"] = "us-east-1"
+    os.environ["AWS_SECRET_ACCESS_KEY"] = "test"  # pragma: allowlist secret
+    os.environ["AWS_ENDPOINT_URL"] = "http://example.com:4566"
+    return FakeS3Client(fake_s3_storage)
+
+@pytest.fixture(scope="function")
+def boto_s3_client(fake_s3_storage) -> S3Client:
```

Comment:
can we remove boto?

Reply:
Most probably. If it's used, it will only be in the integration tests, but the app should use its own boto client anyway.

Reply:
Ah wait. It's plugged as a dependency in the app. Right now it has to be there for integration tests.